# Patch from upstream to ensure wxwidgets support works with cmake 2.8.0

diff -ur plplot-5.9.2-orig/cmake/modules/wxwidgets.cmake plplot-5.9.2/cmake/modules/wxwidgets.cmake
--- plplot-5.9.2.orig/cmake/modules/wxwidgets.cmake	2009-01-18 20:24:12.000000000 +0000
+++ plplot-5.9.2/cmake/modules/wxwidgets.cmake	2010-01-18 23:52:40.000000000 +0000
@@ -60,12 +60,21 @@
     wxwidgets_COMPILE_FLAGS
     "-I${wxWidgets_INCLUDE_DIRS}"
     )
-  # For case (cvs version of CMake as of 2008-03-23, but not cmake-2.4.8)
-  # when wxWidgets_DEFINITIONS is a list.
+  string(REGEX REPLACE ";" " -D"
+    wxWidgets_DEFINITIONS_wD
+    "-D${wxWidgets_DEFINITIONS}"
+    )
+        set(wxWidgets_DEFINITIONS_DEBUG_wD "")
+  if(wxWidgets_DEFINITIONS_DEBUG)
+         string(REGEX REPLACE ";" " -D"
+            wxWidgets_DEFINITIONS_DEBUG_wD
+            "-D${wxWidgets_DEFINITIONS_DEBUG}"
+            )
+  endif(wxWidgets_DEFINITIONS_DEBUG)
   string(REGEX REPLACE ";" " " 
     wxwidgets_COMPILE_FLAGS
     ${wxwidgets_COMPILE_FLAGS}
-    " ${wxWidgets_DEFINITIONS}"
+    " ${wxWidgets_DEFINITIONS_wD} ${wxWidgets_DEFINITIONS_DEBUG_wD}"
     )
   # Convert wxWidgets_LIBRARIES to full pathname form.
   cmake_link_flags(wxwidgets_LINK_FLAGS "${wxWidgets_LIBRARIES}")
diff -ur plplot-5.9.2.orig/bindings/wxwidgets/CMakeLists.txt plplot-5.9.2/bindings/wxwidgets/CMakeLists.txt
--- plplot-5.9.2.orig/bindings/wxwidgets/CMakeLists.txt	2009-01-18 20:24:12.000000000 +0000
+++ plplot-5.9.2/bindings/wxwidgets/CMakeLists.txt	2010-01-18 23:55:55.000000000 +0000
@@ -55,9 +55,18 @@
   endif(BUILD_SHARED_LIBS)
 
   INCLUDE_DIRECTORIES( ${wxWidgets_INCLUDE_DIRS} )
-  ADD_DEFINITIONS( ${wxWidgets_DEFINITIONS} )
 
-  target_link_libraries(
+  IF(wxWidgets_DEFINITIONS)
+    SET_PROPERTY(DIRECTORY APPEND
+      PROPERTY COMPILE_DEFINITIONS ${wxWidgets_DEFINITIONS})
+  ENDIF(wxWidgets_DEFINITIONS)
+
+  IF(wxWidgets_DEFINITIONS_DEBUG)
+    SET_PROPERTY(DIRECTORY APPEND
+      PROPERTY COMPILE_DEFINITIONS_DEBUG ${wxWidgets_DEFINITIONS_DEBUG})
+  ENDIF(wxWidgets_DEFINITIONS_DEBUG)
+
+target_link_libraries(
   plplotwxwidgets${LIB_TAG}
   plplotcxx${LIB_TAG} 
   ${wxwidgets_LINK_FLAGS}
@@ -103,9 +112,14 @@
     endif(LIB_TAG)
     
     # Determine libplplotwxwidgets${LIB_TAG}_COMPILE_FLAGS
-    STRING(REGEX REPLACE ";" " "
+    set(wxWidgets_DEFINITIONS_all "${wxWidgets_DEFINITIONS}")
+    if(wxWidgets_DEFINITIONS_DEBUG)
+          set(wxWidgets_DEFINITIONS_all "${wxWidgets_DEFINITIONS};${wxWidgets_DEFINITIONS_DEBUG}")
+    endif(wxWidgets_DEFINITIONS_DEBUG)
+
+    STRING(REGEX REPLACE ";" " -D"
     libplplotwxwidgets${LIB_TAG}_COMPILE_FLAGS
-    "${wxWidgets_DEFINITIONS}"
+    "-D${wxWidgets_DEFINITIONS_all}"
     )
     STRING(REGEX REPLACE ";" " -I"
     wxWidgets_IFLAGS
